### PR TITLE
Add splunkindex grain

### LINKED ISSUE #107: Add splunk index name to grains
This is a pretty splunk-specific feature but would be very useful for the topfile features in #106. Team-specific targeting and the like.

### PR DESCRIPTION
**Pull request by mew1033**
Not sure if this is how you were thinking of adding the grain, but this should fix #107.

### CODE REVIEW

**Comment by mew1033** [comment]
Also, this only supports the new style config.

**Comment by basepi** [comment]
Have you tested this? I want to make sure there's no gotchas in terms of the config.get usage.

**Comment by fossam** [comment]
Besides the code, we have a choice here - we are going to depend on this grain on top files to do  matching, so the top files may have to always be specific to returner or at least depend on it. The choice is to either to accept that dependency, or have a more generic grain that will leave room for other returners.

I personally would lean towards more generic approach. For example, for testing purposes would be great to have a returner that just writes to plain files and it would be neat not to change top files for every test or use case like that. However, these are my first on thoughts on the matter and it could be that I am overlooking relevant details that make this less of an issue.

**Comment by basepi** [comment]
I mean, we're only going to use the index for targeting on our internal top.sls file. And all internal hosts must use the splunk returner. So I think it's OK to introduce that dependency.

File: hubblestack/extmods/grains/splunkindex.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+
+def splunkindex():
+    '''
+    Return splunk index from config file in grain
+    '''
+    # Provides:
+    #   splunkindex
+    grains = {}
+    index = __salt__['config.get']('hubblestack:returner:splunk:index', default='unknown')
```
Comment (basepi) [inline hubblestack/extmods/grains/splunkindex.py]: Let's either default this to `None`, or just not define it at all if we can't find that value.

Comment (fossam) [inline hubblestack/extmods/grains/splunkindex.py]:
```
if __salt__['config.get']('hubblestack:returner:splunk:index')
        grains['splunkindex'] = index
return grains
```

**Comment by mew1033** [comment]
I'm going to change this up a bit to support custom grains in the config.

**Comment by mew1033** [comment]
@basepi @fossam Alright, I changed this to allowing any custom grains defined in config.

File: hubblestack/extmods/grains/configgrains.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+
```
Comment (basepi) [inline hubblestack/extmods/grains/configgrains.py]: Can you just add a docstring with sample config? This looks great otherwise.

Comment (mew1033) [reply hubblestack/extmods/grains/configgrains.py]: Done

**Pull request merged**